src/interface/ICrosschain.sol

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+interface ICrosschain {
+
+    function sendCrossChainTransaction(
+        uint64 _destinationChain,
+        address _callAddress,
+        address _recipient,
+        address _token,
+        uint256 _amount,
+        bytes calldata _data,
+        bytes calldata _extraArgs
+    ) external;
+
+    function setBridge(address _bridge) external;
+    function getBridge(address _bridge) external;
```

Comment:
`function getBridge() view returns (address)`?

Reply:
We'll probably need a different naming than `Bridge` because this design captures both cross-chain messaging and bridging. Bridges work using cross-chain messaging anyways. 
We should probably call it a `Relay` or `Router`?